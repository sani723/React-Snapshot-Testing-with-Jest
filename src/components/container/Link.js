import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Status from '../../utils/Constants';
import './Link.css';

class Link extends Component {

  constructor(props) {
    super(props);

    this.state = {
      class: Status.NORMAL
    };

  } // constructor ends

  handleMouseEnter = () => {
    this.setState({
      class: Status.HOVERED
    });
  }

  handleMouseLeave = () => {
    this.setState({
      class: Status.NORMAL
    });
  }

  render() {

    return (
      <a
        className={this.state.class}
        href={this.props.page}
        onMouseEnter={this.handleMouseEnter}
        onMouseLeave={this.handleMouseLeave}
      >
        {this.props.text}
      </a>
    );

  } // render ends

}

Link.defaultProps = {
  page: "#",
  text: "Follow us"
};

Link.propTypes = {
  page: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired
};

export default Link;
