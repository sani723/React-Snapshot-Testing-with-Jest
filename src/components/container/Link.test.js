import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';

import Link from './Link';

configure({ adapter: new Adapter() });

describe("<Link />", () => {
  it("It should render as expected", () => {
    const component = shallow(<Link />);
    let tree = toJson(component);
    expect(tree).toMatchSnapshot();

    tree.props.onMouseEnter();
    tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
