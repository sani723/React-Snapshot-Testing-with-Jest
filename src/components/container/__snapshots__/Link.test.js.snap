// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> It should render as expected 1`] = `
<a
  className="normal"
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Follow us
</a>
`;

exports[`<Link /> It should render as expected 2`] = `
<a
  className="normal"
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Follow us
</a>
`;
